sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -81,19 +81,20 @@ def __init__(
     ) -> None:
         self.client_connection = client_connection
         self.id = id
-        self._properties = properties
         self.database_link = database_link
         self.container_link = "{}/colls/{}".format(database_link, self.id)
         self._is_system_key: Optional[bool] = None
         self._scripts: Optional[ScriptsProxy] = None
+        if properties:
+            self.client_connection.collection_properties_cache[self.container_link] = properties
 
     def __repr__(self) -> str:
         return "<ContainerProxy [{}]>".format(self.container_link)[:1024]
 
     async def _get_properties(self) -> Dict[str, Any]:
-        if self._properties is None:
-            self._properties = await self.read()
-        return self._properties
+        if self.container_link not in self.client_connection.collection_properties_cache:
+            self.client_connection.collection_properties_cache[self.container_link] = await self.read()
```

Comment:
```suggestion
            await self.read()
```
the `read()` method is already updating the cache, so we don't need to do it here as well.